### DISCUSSION TITLE
any way to migrate a resolver's type from Int to CustomInt without breaking old clients?

### DISCUSSION THREAD

**Discussion by shawnjones253** [MichalLytek/type-graphql#1656] [Q&A] [answered]
I've got several resolvers that use this signature:

```
query MyQuery($take: Int) {
  findMany(take: $take) {
    id
  }
}
```

Now I want to start using a custom scalar, which is also an Int, but does some other internal validation (max value for example). So I create a `MinMaxScalar` as a custom `GraphQLScalarType`, and change the `@Field((_type) => Int)` to `@Field((_type) => MinMaxScalar)`.

Is there any way to do this without breaking old clients? It seems since graphql doesn't have input unions of scalar types, the only choice is to rename the field during the migration, is that right?

**Comment by MichalLytek** [comment, marked as answer]
That's a general GraphQL question. You may look for a better approach on stackoverflow or some Discord channels.
